apps/explorer/lib/explorer/etherscan/logs.ex

```diff
@@ -165,6 +161,19 @@ defmodule Explorer.Etherscan.Logs do
           union: ^internal_transaction_log_query
         )
 
+      dynamic =
+        dynamic(
+          [transaction],
+          ^Transaction.where_transactions_to_from(address_hash) and
+            transaction.block_number >= ^prepared_filter.from_block and
+            transaction.block_number <= ^prepared_filter.to_block
+        )
+
+      all_transaction_logs_query =
+        all_transaction_logs_query_base
+        |> where([transaction], transaction.created_contract_address_hash == ^address_hash)
+        |> or_where([transaction], ^dynamic)
+
```

Comment:
Can we simplify it this way to get rid of the duplicated block range condition? 
```suggestion
      dynamic =
        dynamic(
          [transaction],
          ^Transaction.where_transactions_to_from(address_hash) or
            transaction.created_contract_address_hash == ^address_hash
        )

      all_transaction_logs_query =
        all_transaction_logs_query_base
        |> where([transaction], ^dynamic)
```

Reply:
Yes, I think, this should work. Thanks.